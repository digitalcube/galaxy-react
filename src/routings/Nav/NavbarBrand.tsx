import React , {FC} from 'react'
import { NavbarBrand as ReactstrapNavbarBrand } from 'reactstrap'
import { BrandName, Logo } from '../../components'
import { useLinkTag } from '../../providers'


export type NavbarBrandProps = {
    name: BrandName;
    width?: number;
    href?: string;
}

export const NavbarBrand: FC<NavbarBrandProps> = ({name, width = 24, href = '/'}) => {
    const { tag } = useLinkTag()
    return (
        <ReactstrapNavbarBrand tag={tag} href={href}>
            <Logo name={name} width={width} />
        </ReactstrapNavbarBrand>
    )
}