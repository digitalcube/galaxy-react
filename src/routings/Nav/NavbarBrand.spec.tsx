import React from 'react';
import { render } from '@testing-library/react'
import { Default as NavbarBrand } from './NavbarBrand.stories';

describe('NavbarBrand', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<NavbarBrand {...{
        name: 'shifter',
        ...NavbarBrand.args,
    }}/>);
    expect(asFragment()).toMatchSnapshot()
  });
});
