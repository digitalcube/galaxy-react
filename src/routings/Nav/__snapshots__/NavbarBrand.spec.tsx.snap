// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarBrand to match snapshot 1`] = `
<DocumentFragment>
  <a
    class="navbar-brand"
    href="/"
  >
    <img
      alt="Shifter"
      height="32"
      width="24"
    />
  </a>
</DocumentFragment>
`;
